includes/class-wc-gateway-swedbank-pay-invoice.php

```diff
@@ -266,267 +307,126 @@ public function thankyou_page( $order_id ) {
 		} catch ( \Exception $e ) {
-			$this->log( sprintf( '[ERROR] Create Approved Legal Address: %s', $e->getMessage() ) );
 			wc_add_notice( $e->getMessage(), 'error' );
 
 			return false;
 		}
 
-		// Save legal address
-		$legal_address = $result['approvedLegalAddress'];
-		update_post_meta( $order_id, '_payex_legal_address', $legal_address );
-
-		// Transaction Activity: FinancingConsumer
-		try {
-			$params = [
-				'transaction'  => [
-					'activity' => 'FinancingConsumer'
-				],
-				'consumer'     => [
-					'socialSecurityNumber' => $ssn,
-					'customerNumber'       => $user_id,
-					'email'                => $email,
-					'msisdn'               => '+' . ltrim( $phone, '+' ),
-					'ip'                   => swedbank_pay_get_remote_address()
-				],
-				'legalAddress' => [
-					'addressee'     => $legal_address['addressee'],
-					'coAddress'     => $legal_address['coAddress'],
-					'streetAddress' => $legal_address['streetAddress'],
-					'zipCode'       => $legal_address['zipCode'],
-					'city'          => $legal_address['city'],
-					'countryCode'   => $legal_address['countryCode']
-				]
-			];
-
-			$result = $this->request( 'POST', $create_authorize_href, $params );
-		} catch ( \Exception $e ) {
-			$this->log( sprintf( '[ERROR] Create Authorize: %s', $e->getMessage() ) );
-			wc_add_notice( $e->getMessage(), 'error' );
-
-			return false;
-		}
-
-		return [
+		return array(
 			'result'   => 'success',
-			'redirect' => $this->get_return_url( $order )
-		];
+			'redirect' => $this->get_return_url( $order ),
+		);
 	}
 
 	/**
-	 * Get Order Info
+	 * Process Refund
 	 *
-	 * @param WC_Order $order
+	 * If the gateway declares 'refunds' support, this will allow it to refund
+	 * a passed in amount.
 	 *
-	 * @return array
+	 * @param int $order_id
+	 * @param float $amount
+	 * @param string $reason
+	 *
+	 * @return  bool|wp_error True or false based on success, or a WP_Error object
 	 */
-	protected function get_order_info( $order ) {
-		$amount       = 0;
-		$vatAmount    = 0;
-		$descriptions = [];
-		$items        = $this->get_order_items( $order );
-		foreach ( $items as $item ) {
-			$amount         += $item['price_with_tax'];
-			$vatAmount      += $item['tax_price'];
-			$unit_price     = sprintf( "%.2f", $item['price_without_tax'] / $item['qty'] );
-			$descriptions[] = [
-				'product'    => $item['name'],
-				'quantity'   => $item['qty'],
-				'unitPrice'  => (int) round( $unit_price * 100 ),
-				'amount'     => (int) round( $item['price_with_tax'] * 100 ),
-				'vatAmount'  => (int) round( $item['tax_price'] * 100 ),
-				'vatPercent' => sprintf( "%.2f", $item['tax_percent'] ),
-			];
+	public function process_refund( $order_id, $amount = null, $reason = '' ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return false;
 		}
 
-		return [
-			'amount'     => $amount,
-			'vat_amount' => $vatAmount,
-			'items'      => $descriptions
-		];
+		// Full Refund
+		if ( is_null( $amount ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_null() is discouraged; use strict comparison "=== null" instead.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)